admin_frontend/src/modals/CoingeckoId.jsx

```diff
@@ -0,0 +1,215 @@
+import { useEffect, useState } from "react";
+import AddCircleIcon from "@mui/icons-material/AddCircle";
+import RemoveCircleIcon from "@mui/icons-material/RemoveCircle";
+import toast from "react-hot-toast";
+import { TextField } from "@mui/material";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import Typography from "@mui/material/Typography";
+import Modal from "@mui/material/Modal";
+
+const style = {
+	width: "100%",
+	bgcolor: "background.paper",
+	border: "2px solid #000",
+	boxShadow: 24,
+	p: 2,
+	flexDirection: "column",
+	overflowX: "scroll",
+};
+
+const defaultSupportedNetworksRow = {
+	chainId: "",
+	coingeckoId: "",
+};
+
+const CoingeckoIdModal = ({
+	supportedNetworks,
+	setSupportedNetworks,
+	open,
+	handleClose,
+}) => {
+	const [supportedNetworkRow, setSupportedNetworkRow] = useState(
+		defaultSupportedNetworksRow
+	);
+	const [ids, setIds] = useState([]);
+
+	const addRow = () => {
+		const tempId = ids;
+		if (
+			supportedNetworkRow.coingeckoId === "" ||
+			supportedNetworkRow.chainId === "" ||
+			supportedNetworkRow.chainId === 0
+		) {
+			toast.error("Please fill all textfields");
+			return;
+		}
+		if (
+			!ids.find(
+				(network) =>
+					network.chainId === supportedNetworkRow.chainId &&
+					network.coingeckoId === supportedNetworkRow.coingeckoId
+			)
+		) {
+			if (!supportedNetworks[supportedNetworkRow.chainId.toString()])
+				supportedNetworks[supportedNetworkRow.chainId.toString()] = [
+					supportedNetworkRow.coingeckoId,
+				];
+			else {
+				supportedNetworks[supportedNetworkRow.chainId.toString()].push(
+					supportedNetworkRow.coingeckoId
+				);
+			}
+			tempId.push({
+				chainId: supportedNetworkRow.chainId,
+				coingeckoId: supportedNetworkRow.coingeckoId,
+			});
+			setIds(tempId);
+			setSupportedNetworks(supportedNetworks);
+			setSupportedNetworkRow(defaultSupportedNetworksRow);
+			console.log("ids: ", ids, supportedNetworks);
+		} else {
+			toast.error("ChainId Already present");
+		}
+	};
+
+	const deleteRow = (network) => {
+		setIds(
+			ids.filter(
+				(element) =>
+					element.chainId !== network.chainId &&
+					element.coingeckoId !== network.coingeckoId
+			)
+		);
+		setSupportedNetworks(
+			supportedNetworks[network.chainId.toString()].filter(
+				(element) => element !== network.coingeckoId
+			)
+		);
+		setSupportedNetworkRow(defaultSupportedNetworksRow);
+	};
+
+	useEffect(() => {
+		const coingeckoIds = [];
+		for (const chain in supportedNetworks) {
+			for (const id of supportedNetworks[chain]) {
+				coingeckoIds.push({
+					chainId: chain,
+					coingeckoId: id,
+				});
+			}
+		}
```

Comment:
Could be written using js array helpers like `.map`

Reply:
I tried that but map is async function and for one array I had to get the key and for the other array I had to get the value, it was going all over the place and didnt set the data correctly